final_task/rss_reader/get_cache.py

```diff
@@ -0,0 +1,102 @@
+from dominate.tags import div, h2, img, p, link
+from datetime import datetime
+import dominate
+import psycopg2
+import base64
+import time
+import json
+
+from . import logg, converter, news_parser, topdf
+
+
+def dateToStamp(arg_date):
+    '''
+    1. convert --date into timestamp
+    '''
+    arg_date = str(arg_date)
+    arg_date = time.mktime(datetime.strptime(arg_date, '%Y%m%d').timetuple())
+    return arg_date
+
+
+def getCacheFromDB(arg_date):
+    '''
+    1. connect to database
+    2. select from table news with published date equals --date
+    3. convert news into html or pdf if there are --tohtml or --topdf arguments
+    3. or print news in stdout
+    '''
+    try:
+        with psycopg2.connect(database="postgres",user='postgres',password='rssreader',host='localhost',port='5432') as con:
+            with con.cursor() as cur:
+                cur.execute('''SELECT title, link, image, description FROM news WHERE           pub_date_stamp >= %s and pub_date_stamp < %s''',
+                            (dateToStamp(arg_date), dateToStamp(int(arg_date) + 1)))
+
+                records = cur.fetchall()
+                return records
+    except psycopg2.ProgrammingError as e:
+        print("psycopg2.ProgrammingError: " + str(e))
+        logg.logging.error(str(e))
+    finally:
+        con.close()
```

Comment:
This may potentially cause `UnboundLocalError` when variable referenced before assignment.
This is possible in case we will fail to connect to DB for some reason.

It is better to do as stated in docs:
```
con = psycopg2.connect....
try:
    ...
finally:
    con.close()
```

Same for another similar situation in `news_parser` file.